### PR TITLE
Poll and group scan targets - ec2

### PR DESCRIPTION
**Pull request by belosh59**

### CODE REVIEW

**Comment by belosh59** [comment]
@kickroot  JFYI verification of bulkhead limitation over the shared ExecutorService between all lambdas
**Preconditions**:
1. We have VPC with about 4 EC2 Target services
2. We configured semaphore to allow only 2 active scans per lambda for EC2 Targets
```
--  Group 2     "i-02d707a66b2c4d632"	"subnet-022a263e58d1ada34"	"sg-07a6077c9af3c6801"
--                     "i-0bcfc92093c876165"	"subnet-022a263e58d1ada34"	"sg-07a6077c9af3c6801"
--                     "i-0c4779103ad377ecb"	"subnet-022a263e58d1ada34"	"sg-07a6077c9af3c6801"
--                     "i-0c4779103ad377ecb"	"subnet-022a263e58d1ada34"	"sg-07a6077c9af3c6801"
```

**Expected result:**
- First 2 EC2-threads for particular lambda in that VPC should start immediately
- The 3rd EC2-thread `i-0c4779103ad377ecb` **will wait for some time** while previous `i-0bcfc92093c876165` thread do not release the semaphore

Screenshot 2021-07-22 at 21 24 26

**Comment by belosh59** [comment]
Final state after lates changes and essential refactoring:

Screenshot 2021-07-23 at 02 07 43

File: magpie-core/pom.xml [hunk 1]
```diff
@@ -55,6 +56,18 @@
       <scope>compile</scope>
     </dependency>
 
+    <!-- TODO: design - are we able to avoid wiring AWS into the core package? magpie-aws / magpie-dmap plugin? -->
```
Comment (kickroot) [inline magpie-core/pom.xml:59]: Remind me why this is required in `core`?

File: magpie-core/src/main/java/io/openraven/magpie/core/dmap/client/DMapMLClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+
+public class DMapMLClientImpl implements DMapMLClient {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DMapMLClientImpl.class);
+  private static final String DMAP_ML_SERVICE_URL = "https://api.openraven.com/dmap-ml/predict-annotated";
```
Comment (kickroot) [inline magpie-core/src/main/java/io/openraven/magpie/core/dmap/client/DMapMLClientImpl.java:21]: Can we pull this out to a configuration option in `config.yaml`?

**Review by kickroot — APPROVED** [review]

File: magpie-core/pom.xml (hunk 1 above)
Comment (belosh59) [reply magpie-core/pom.xml:59]: We need this due to the AWS Clients invocation under the core module for Role/Policy creation and Lambda creation/invocation/deletion

Comment (kickroot) [reply magpie-core/pom.xml:59]: Makes sense.  At some point we should make a `dmap` module and move it to there, but we'll plan that for a future release.

**Review by kickroot — APPROVED** [review]

**Pull request merged**